src/test/java/com/cinema/services/HallServiceImplTest.java

```diff
@@ -0,0 +1,29 @@
+package com.cinema.services;
+
+import com.cinema.controllers.rest.HallRestController;
+import org.junit.jupiter.api.Test;
+
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class HallServiceImplTest {
+    @LocalServerPort
+    int port;
+    @Autowired
+    HallRestController hallRestController;
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @Test
+    public void save() {
+        assertThat(this.restTemplate.getForObject("http://localhost:" + port + "/", String.class)).contains("Hello");
+    }
```

Comment:
did you try without port? as per doc

import org.junit.Test;
import org.junit.runner.RunWith;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.test.context.junit4.SpringRunner;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringRunner.class)
@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
public class RandomPortTestRestTemplateExampleTests {

	@Autowired
	private TestRestTemplate restTemplate;

	@Test
	public void exampleTest() {
		String body = this.restTemplate.getForObject("/", String.class);
		assertThat(body).isEqualTo("Hello World");
	}

}

Reply:
https://docs.spring.io/spring-boot/docs/2.1.5.RELEASE/reference/html/boot-features-testing.html

Reply:
I tried without port and evrithing works.